### PR TITLE
add dead_letter_policy to pubsub subscriptions

### PR DESCRIPTION
**Pull request by devops-dmitry**

### CODE REVIEW

File: modules/gcp/pubsub/input.tf [hunk 1]
```diff
@@ -51,6 +51,8 @@ variable "subscriptions" {
     ttl                          = optional(string, "2678400s")
     minimum_backoff              = optional(string, null)
     maximum_backoff              = optional(string, null)
+    dead_letter_topic            = optional(string, null)
+    max_delivery_attempts        = optional(number, null)
```
Comment (ggramal) [inline modules/gcp/pubsub/input.tf:55]: Max delivery attempts for what? Maybe we should introduce an object var `dead_letter` with attributes like topic and max_delivery_attemps

Comment (ggramal) [reply modules/gcp/pubsub/input.tf]: well `var.subscription["xyz"].dead_letter.max_delivery_attempts` seems a much cleaner solution. In situations like this https://github.com/ggramal/tf-modules/blob/0e9c86ea7248ce09c53e7e71ca9c6e1d95cf12f1/modules/gcp/pubsub/main.tf#L33 it also seems to help (you can just compare dead_letter to null)

Comment (devops-dmitry) [reply modules/gcp/pubsub/input.tf]: now we have:
`dead_letter_topic = "projects/${local.gcp_project.name }/topics/topic_name"`
will be used instead:
```
dead_letter = {
  dead_letter_topic = "projects/${local.gcp_project.name}/topics/protect_dead_letter"
}
```

Comment (ggramal) [reply modules/gcp/pubsub/input.tf]: >then we will have to write a lot of additional text.
Not sure i understand you here - What additional text we should rewrite?

>complicate the input values by wrapping them with an additional wrapper
:thinking: i think we are just changing its type no?

for me this (NOTE: you dont have to add dead_letter prefix everywhere) 
```
  dead_letter = optional(object(
     {
       topic = string
       max_delivery_attempts = number
     }
  ))
```
is way more cleaner and scalable than this
```
    dead_letter_topic                 = optional(string, null)
    dead_letter_max_delivery_attempts = optional(number, null)
```
Besides that  a subscription can have multiple deadletter topics. So i the solution that you are proposing its very hard to implement with object we can just wrap it in map eg
```
  dead_letter = optional(
    map(
      object(
        {
          topic = string
          max_delivery_attempts = number
        }
      )
    )
  )
```
so with
```
    dead_letter_topic                 = optional(string, null)
    dead_letter_max_delivery_attempts = optional(number, null)
```
we introduce a tech debt and i really dont think we should go with it considering the fact that fix is very easy

Comment (ggramal) [reply modules/gcp/pubsub/input.tf]: ok figured out that we cant set multiple dead letter topics for pubsub + Mr @devops-dmitry convinced me that we should stick with none object approach and see how it goes

**Review by ggramal — APPROVED** [review]

**Pull request merged**